pkg/api/v1alpha1/vspheremachineconfig_webhook.go

```diff
@@ -41,8 +41,11 @@ var _ webhook.Validator = &VSphereMachineConfig{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *VSphereMachineConfig) ValidateCreate() error {
```

Comment:
I only added this on the ValidateCreate webhook because the `Template` field is immutable on update.